tests/devices/qubit/test_measure.py

```diff
@@ -111,22 +164,89 @@ def test_sum_expval_eigs(self, obs, expected):
         res = simulate(qs)
         assert np.allclose(res, expected)
 
-    def test_correct_measurement_used(self):
-        """Test that the correct internal function is used for a given measurement process."""
-        # Test a case where state_measurement_process is used
-        mp1 = qml.state()
-        assert get_measurement_function(mp1) == state_measurement_process
-
-        # Test cases where state_diagonalizing_gates is used
-        mp2 = qml.var(qml.PauliZ(0))
-        assert get_measurement_function(mp2) == state_diagonalizing_gates
-        mp3 = qml.expval(qml.sum(qml.PauliZ(0), qml.PauliX(0)))
-        assert get_measurement_function(mp3) == state_diagonalizing_gates
-        mp4 = qml.expval(qml.sum(*(qml.PauliZ(i) for i in range(8))))
-        assert get_measurement_function(mp4) == state_diagonalizing_gates
-
-        # Test cases where state_hamiltonian_expval is used
-        mp5 = qml.expval(qml.Hamiltonian([], []))
-        assert get_measurement_function(mp5) is state_hamiltonian_expval
-        mp6 = qml.expval(qml.sum(*(qml.prod(qml.PauliY(i), qml.PauliZ(i + 1)) for i in range(7))))
-        assert get_measurement_function(mp6) is state_hamiltonian_expval
+
+class TestSumOfTermsDifferentiability:
+    @staticmethod
+    def f(scale, n_wires=10, offset=0.1, convert_to_hamiltonian=False):
+        ops = [qml.RX(offset + scale * i, wires=i) for i in range(n_wires)]
+
+        t1 = 2.5 * qml.prod(*(qml.PauliZ(i) for i in range(n_wires)))
+        t2 = 6.2 * qml.prod(*(qml.PauliY(i) for i in range(n_wires)))
+        H = t1 + t2
+        if convert_to_hamiltonian:
+            H = H._pauli_rep.hamiltonian()  # pylint: disable=protected-access
+        qs = qml.tape.QuantumScript(ops, [qml.expval(H)])
+        return simulate(qs)[0]
+
+    @staticmethod
```

Comment:
seemed unused
```suggestion
```

Reply:
Just makes it a little easier to copy-paste `expected` into a notebook.  Other wise I would had have to manually remove the `self` input each time.